### PR TITLE
add tests for 'template literal revision'

### PR DESCRIPTION
**Pull request by dnalborczyk**
'template literal revision' (stage 3)

tests are based on ES6 template literal and the draft, significance: small.

https://github.com/tc39/proposals
https://github.com/tc39/proposal-template-literal-revision

### CODE REVIEW

**Comment by afmenez** [comment]
Besides the merge conflict (which seems trivial to fix), is there another issue holding this back?

**Comment by ljharb** [comment]
The original fork has been deleted, so it's not trivial to fix this PR, sadly. @dnalborczyk, do you have any interest in completing this PR?

**Comment by afmenez** [comment]
I have applied the patch to a local branch, and after updating some fields in the header it seems to be working.  The problem is that the test is failing on Firefox 53 & 54, which claim to have this implemented.

These are the two lines that fail on Firefox:
```
        typeof strings[1] === "undefined" &&
        typeof strings[2] === "undefined" &&
```
If I remove these two lines, the test passes on Firefox 53 and fails on Firefox 51, as expected.  I don't know enough to check if the problem is in the test or in Firefox implementation.

**Comment by afmenez** [comment]
There's a new commit in Firefox 55 Nightly ([bugzilla #1108941](https://bugzilla.mozilla.org/show_bug.cgi?id=1108941)), but the problem stated above persists.

**Comment by ljharb** [comment]
@disnet, could you weigh in on this particular test?

File: data-esnext.js [hunk 1]
```diff
@@ -1686,6 +1686,26 @@ exports.tests = [
...
+{
+  name: 'template literal revision',
+  category: 'candidate (stage 3)',
+  significance: 'small',
+  link: 'https://github.com/tc39/proposal-template-literal-revision',
+  exec: function() {/*
+    function fn(strings, a, b) {
+      return typeof strings[0] === "undefined" &&
+        typeof strings[1] === "undefined" &&
+        typeof strings[2] === "undefined" &&
```
Comment (disnet) [inline data-esnext.js:1698]: `\u{55}` is a legal escape sequence, both before and after this proposal so you shouldn't be asserting it is undefined

**Comment by afmenez** [comment]
Great.  Should I remove the 2 tests and submit a new PR?

**Comment by afmenez** [comment]
I created PR #1046 with an updated patch.

File: data-esnext.js (hunk 1 above)
Comment (afmenez) [reply data-esnext.js:1698]: Do you mean none of the parameters should be checked for undefined, just [0] (which make the test pass on Firefox), or just [0] and [1] (leaving out just the `\u{55} `)?

**Pull request closed without merging**